Initial review

osrjv opened the pull request:
Ran the code through black and isort to have consistent formatting, mostly whitespacing issues.

Adding the rest as review comments.

Commit:
Run black

Commit:
Run isort

osrjv commented on consumer.py:
```diff
@@ -1,29 +1,38 @@
     """
-        Inhuman Insurance, Inc. Artificial Intelligence System robot.
-        Consumes traffic data work items.
+    Inhuman Insurance, Inc. Artificial Intelligence System robot.
```
I think we're phasing out the wording "robot", but not sure what the alternative here is.

MonicaBarna replied:
let's replace with automation in this case

osrjv commented on consumer.py:
```diff
@@ -1,29 +1,38 @@
     process_traffic_data()
 
+
 def process_traffic_data():
```
I know this is based on an existing course structure, but I feel like we could reduce unnecessary complexity by not having a function that just calls one function without any arguments.

MonicaBarna replied:
Oh and I already reduced a bit the number of functions compared to RF.
I'll try to merge a bit more then.

osrjv commented on consumer.py:
```diff
@@ -1,29 +1,38 @@
+
 def process_traffic_data():
     for item in workitems.inputs:
         payload = item.payload
```
Nit: This is only used once, below this line. Could merge them. Especially since later it's referred to with `item.payload` again.

osrjv commented on consumer.py:
```diff
@@ -1,29 +1,38 @@
+        if valid:
             status, return_json = post_traffc_data_to_sales_system(traffic_data)
             if status == 200:
                 item.done
```
I assume this was meant to be a function call? Currently does nothing

MonicaBarna replied:
yes

osrjv commented on consumer.py:
```diff
@@ -1,29 +1,38 @@
+            )
 
 
 def validate_traffic_data(traffic_data):
```
Not visible in the diff, but in general instead of `if condition: return True else: return False` you can just write `return condition`. Maybe this makes more sense in the context of the course, it's up to you.

MonicaBarna replied:
I am not even sure. There weren't many ifs until now. It may be simpler if it is explicit? 
But then again, I kinda hope that they already have a python basic at this point.

osrjv commented on producer.py:
```diff
@@ -1,7 +1,5 @@
-from robocorp import http
-from robocorp import workitems
-
 from RPA.JSON import JSON
```
I'm a bit on the fence if we want to use this library just to load a json file. It's two lines in the standard library.

MonicaBarna replied:
I am fine either way. It is only used for the download in the courses and it is the way we want to go forward. So it is good enough for the courses even if very incomplete.

MonicaBarna replied:
The reason was that we want to guide our users to first use our libraries and if we have nothing to go searching in the outside world.

osrjv commented on producer.py:
```diff
@@ -16,27 +14,35 @@
+    Produces traffic data work items.
     """
-    #download_traffic_data()
+    # download_traffic_data()
```
I assume shouldn't be commented out?

MonicaBarna replied:
Yes, I forgot about that. Commented it out while working without internet.

osrjv reviewed (commented):
A lot of the comments I'd have are related to how we structure courses, and some of that baggage comes from the history of Robot Framework. I like the idea of a more streamlined (or Pythonic) version of this, but I can't shake the feeling that it would be nice if we could build the courses around them from the get-go. Not something we need to fix immediately though.

MonicaBarna committed:
Functions regrouping

I can add the extra mention at if to directly return the check result

Also, can explain to directly return the result of something instead of saving it into a var and than returning that.
